### DISCUSSION TITLE
Implement "focus follows mouse"

### DISCUSSION THREAD

**Discussion by pg83** [YaLTeR/niri#144] [Ideas]
Or maybe I miss something from documentation?

**Comment by YaLTeR** [comment]
I'm not sure this is a good idea, since focusing a window will scroll it into view. Imagine with two 67% windows moving your mouse around the screen causing wild scrolling left and right.

**Reply by pg83** [reply]
Of course, I do not propose to make this behavior a default, but to make it under opt in.

In general, I don't really understand what problem you are talking about in the example you gave, it seems to me that it will work exactly as expected by the user who turned on this setting.

I regularly use a qtile with a custom layout similar to paperwm (actually, that's why I decided to try niri) - https://github.com/pg83/bayanwm/blob/main/config.py#L15 , and ffm works pretty well there. Just like in sway.

**Reply by YaLTeR** [reply]
I guess could try and see how it works out.

**Reply by YaLTeR** [reply]
What about a case like: you have your mouse on the left side of the screen and work with the keyboard, so your focus is on the right. Some dialog window opens to the right and causes the layout to scroll left. As it finishes scrolling, the mouse ends up on the window in the middle, focusing it, which interrupts your interaction with the newly opened dialog and diverts keyboard input.

**Reply by pg83** [reply]
It seems to me that in existing solutions it works like this: the focus changes only after moving the mouse/touchpad. Therefore, if the layout of the windows has changed, but the mouse was not moving at that moment (there was no user mouse input), then the focus from "ffm" is not applied.

**Comment by jgregoire** [comment]
Awesomewm focus-follows-mouse only triggers on mouse movement. This avoids a lot of weird unintended focus changes.

The view panning as I move my mouse around sounds like a feature to me, not a problem.

I'm usually keyboard only, but when I'm sorting lots of images into folders, I prefer to use the mouse. Having to click into a window before I can click-and-drag is extra effort vs. focus-follows-mouse.

**Reply by YaLTeR** [reply]
> Having to click into a window before I can click-and-drag

You don't need to focus a window to do that (or to scroll it for that matter).

**Comment by YaLTeR** [comment]
Implemented in df9d721f7462a7edaacb9de4530580a68ab6e8ab.
